scripts/ci/check-success-job.sh

```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+# Check that $SUCCESS_JOB depends on all other jobs in the given $YAML
+
+set -euf -o pipefail
+
+YAML=$1
+SUCCESS_JOB=$2
+
+yq '... comments="" | .jobs | map(. | key) | .[]' < "$YAML" | grep -v "$SUCCESS_JOB" | sort > all_jobs.txt
+yq "... comments=\"\" | .jobs.$SUCCESS_JOB.needs[]" < "$YAML" | grep -v "$SUCCESS_JOB" | sort > dep_jobs.txt
```

Comment:
I abandoned the approach of reading `${{ needs }}` from the github context, because I found that this doesn't expand to JSON but to the string `"Object"` :grin:

Now the script uses `yq` to read the `needs` from the YAML file, and compare them to the full list of jobs in the file. This has the advantage that it can also be run locally, not just on a Github runner